sqlmesh/core/model/definition.py

```diff
@@ -1882,12 +1919,19 @@ def _python_env(
                                     path,
                                 )
                             used_variables.add(args[0].this.lower())
-                elif macro_func_or_var.__class__ is d.MacroVar:
+                elif isinstance(macro_func_or_var, d.MacroVar):
                     name = macro_func_or_var.name.lower()
                     if name in macros:
                         used_macros[name] = macros[name]
                     elif name in variables:
                         used_variables.add(name)
+                elif (
+                    isinstance(macro_func_or_var, exp.Identifier) and "@" in macro_func_or_var.this
+                ):
+                    for m in MacroStrTemplate.pattern.findall(macro_func_or_var.this):
+                        var_name = m[2] or m[1]
```

Comment:
what's m2 and m1? can you do for x, y so i can read what these things are

Reply:
unpacked the tuple